### ISSUE TITLE
`cupyx.scipy.ndimage.affine_transform` broken with `mode=opencv`

### ISSUE THREAD

**Issue by takagi** [cupy/cupy#3601]
Rel #3559.

`cupyx.scipy.ndimage.affine_transform` can take `opencv` for its `mode` paramter to make it work like OpenCV's one. However, some change (maybe #2483?), broke it and the current master does not work with the parameter.

```
>>> cupyx.scipy.ndimage.affine_transform(im, M, output_shape=smaller_shape, output=smaller, mode='opencv')
/home/ext-mtakagi/cupy/cupyx/scipy/ndimage/interpolation.py:30: UserWarning: In the current feature the default order of affine_transform is 1. It is different from scipy.ndimage and can change in the future.
  'the future.'.format(func_name))
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/home/ext-mtakagi/cupy/cupyx/scipy/ndimage/interpolation.py", line 212, in affine_transform
    integer_output=integer_output)
  File "cupy/util.pyx", line 103, in cupy.util.memoize.decorator.ret
  File "/home/ext-mtakagi/cupy/cupyx/scipy/ndimage/_interp_kernels.py", line 405, in _get_affine_kernel
    integer_output=integer_output,
  File "/home/ext-mtakagi/cupy/cupyx/scipy/ndimage/_interp_kernels.py", line 269, in _generate_interp_custom
    mode, ixvar, 'xsize_{}'.format(j)))
  File "/home/ext-mtakagi/cupy/cupyx/scipy/ndimage/filters.py", line 720, in _generate_boundary_condition_ops
    return ops
UnboundLocalError: local variable 'ops' referenced before assignment
```

**Comment by asmafar** [comment]
Any updates on this issue please ?
Thanks

**Comment by grlee77** [comment]
> Any updates on this issue please ?

I will take a quick look.

**Comment by grlee77** [comment]
This should be fixed by the changes #4130. It should also be possible to backport the fix to the 8.x branch.
